ruby/vampires.rb

```diff
@@ -0,0 +1,131 @@
+		survey_form # Call the method survey_form
+	counter +=1
+	end
+	times
+end
+
+def survey_form
+	puts "What's your name?"
+	employee_name = gets.chomp
+
+	# To compare the answers to see if the informed_age matches the informed_birth:
+	puts "What will be your age at the last day of this year?"
+	informed_age = gets.chomp.to_i #convert input to integer
+
+	puts "What year were you born?"
+	informed_birth = gets.chomp.to_i 
+	age = Date.today.year - informed_birth #get current year 
+
+	puts "Our company cafeteria serves garlic bread. Should we order some for you? (y/n)"
+	wants_garlic = nil # 1.initialize variable to use on loop
+
+	# Create loop to check if input is valid
+	until wants_garlic == 'y' || wants_garlic == 'n'
+		wants_garlic = gets.chomp.downcase #use .downcase to make input "case insensitive"
+		if wants_garlic == 'y' 
+			garlic = true #boolean to validate conditions
+		elsif wants_garlic == 'n'
+			garlic = false
+		else
+			puts 'Option invalid! Please, answer with "y" or "n".' #Alert msg.
+		end
+	end
+
+	p garlic
+
+	puts "Would you like to enroll in the company’s health insurance? (y/n)"
+	wants_health_insurance = nil
+		
+	until wants_health_insurance == 'y' || wants_health_insurance == 'n'	
+		wants_health_insurance = gets.chomp.downcase
+		if wants_health_insurance == 'y' 
+			health_insurance = true
+		elsif wants_health_insurance == 'n'
+			health_insurance = false
+		else
+			puts 'Option invalid! Please, answer with "y" or "n".'
+		end
+	end
+ 	 
+	# Release 4: Check Allergies
+	puts "Name any allergies, one at a time. You should type “done” when finished."
+	allergy = nil
+	is_vampire_detected = false # Boolean to check if vampire is detected through suspect allergies.
+
+	# Loop to name allergies, break if "done". If the allergy is "sunshine", skip to the result of "Probably a vampire."
+	until allergy == "done"
+		allergy = gets.chomp.downcase
+		if allergy == "sunshine"
+			is_vampire_detected = true
+		end
+	end
+
+	# Detect a vampire through allergies
+	if is_vampire_detected 
+		puts "Probably a vampire." 
+	end
+
+	# Case vampire is not detected through allergies:
+	if !is_vampire_detected 
+		# Compare to see if informed age is equal actual age
+		if informed_age == age
+			is_age_correct = true
+		else 
+			is_age_correct = false
+		end
+
+		#Vampire Detecting logic:
+
+		vampire_level = nil #variable to be updated so that the result can be based on the latest condition matched.
+
+		#Evaluating the conditions:
+		if is_age_correct && ( garlic || health_insurance )
+			vampire_level = 0
+		end
+
+		if !is_age_correct && ( !garlic || !health_insurance )
+			vampire_level = 1
+		end
+
+		if !is_age_correct && !garlic && !health_insurance
+			vampire_level = 2
+		end
+
+		if employee_name == "Drake Cula" || employee_name == "Tu Fang"
+			vampire_level = 3
+		end
+	
+		#Analyzing Vampire Detection result through vampire_level:
+		if vampire_level == 0
+			puts "Probably not a vampire."
```

Comment:
Watch out with `puts` vs `p`  puts is an interesting method, it takes an object as an argument, displays it to `$STDOUT` (your terminal), and returns nil. The thing to remember with `puts` is that it returns `nil` after it is called, so when we write code like
```
def add
	puts 1 + 2
end
```
we think we are returning 3 when we call the method because it displays to our screen, but we actually get nil (as the the last evaluated expression is the return of a method).
To avoid this, just omit puts from the method. Now you may say, 'But how do we know it's working?' Great question! For starters we could call our method and display its return with the more friendly development-only method `p` like so.
	```p add```
The gist is, only use puts when you want to display text to the screen, if your methods are supposed to do something else, let them and if you are curious look at the results outside of that method.